cli/azd/test/functional/cli_test.go

```diff
@@ -234,6 +235,54 @@ func Test_CLI_ProvisionCache(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func Test_CLI_ProvisionStateWithDown(t *testing.T) {
+	t.Setenv("AZURE_RECORD_MODE", "live")
+	ctx, cancel := newTestContext(t)
+	defer cancel()
+
+	dir := tempDirWithDiagnostics(t)
+	t.Logf("DIR: %s", dir)
+
+	session := recording.Start(t)
+
+	envName := randomOrStoredEnvName(session)
+	t.Logf("AZURE_ENV_NAME: %s", envName)
+
+	cli := azdcli.NewCLI(t, azdcli.WithSession(session))
+	cli.WorkingDirectory = dir
+	cli.Env = append(cli.Env, os.Environ()...)
+	cli.Env = append(cli.Env, "AZURE_LOCATION=eastus2")
+
+	err := copySample(dir, "storage")
+	require.NoError(t, err, "failed expanding sample")
+
+	_, err = cli.RunCommandWithStdIn(ctx, stdinForInit(envName), "init")
+	require.NoError(t, err)
+
+	expectedOutputContains := "There are no changes to provision for your application."
+
+	initial, err := cli.RunCommandWithStdIn(ctx, stdinForProvision(), "provision")
+	require.NoError(t, err)
+	require.NotContains(t, initial.Stdout, expectedOutputContains)
+
+	// Second provision should use cache
+	secondProvisionOutput, err := cli.RunCommandWithStdIn(ctx, stdinForProvision(), "provision")
+	require.NoError(t, err)
+	require.Contains(t, secondProvisionOutput.Stdout, expectedOutputContains)
+
+	// down to delete resources
+	_, err = cli.RunCommandWithStdIn(ctx, "", "down", "--force", "--purge")
+	require.NoError(t, err)
+
+	// use flag to force provision
```

Comment:
NIT: Isn't it more correct to say // run provision and make sure it does not skip infrastructure provisioning?